language/borrow-graph/src/graph.rs

```diff
@@ -294,9 +310,46 @@ impl<Loc: Copy, Lbl: Clone + Ord> BorrowGraph<Loc, Lbl> {
                 }
             }
         }
+        debug_assert!(joined.check_invariant());
         joined
     }
 
+    //**********************************************************************************************
+    // Consistency/Invariants
+    //**********************************************************************************************
+
+    fn check_invariant(&self) -> bool {
```

Comment:
Looks great.  Take a look at how I was using the mirai annotations in my borrow graph implementaiton.

@hermanventer : FYI.